chapters/en/slides/module7/module7_03.md

```diff
@@ -0,0 +1,389 @@
+
+But since we can click the legend to show the points from only a
+specific genre, this problem is alleviated.
+
+---
+
+## Binding selections to dropdowns instead of legends
+
+``` python
+genres = sorted(movies['Major_Genre'].unique())
+dropdown = alt.binding_select(name='Genre ', options=genres)
+select_genre = alt.selection_single(fields=['Major_Genre'], bind=dropdown)
+
+points.add_selection(select_genre).encode(
+    opacity=alt.condition(select_genre, alt.value(0.7), alt.value(0.1)))
+```
+
+<iframe src="/module7/charts/03/unnamed-chunk-3.html" width="100%" height="420px" style="border:none;">
+</iframe>
+
+Notes: Driving our interactions via a dropdown instead of a legend could
+be useful if we have too many categories to show in the legend, or if we
+wanted to have a different dataframe column in the dropdown and in the
+legend.
+
+Here we use the same column as for coloring the points, so that you can
+clearly see how the interaction works.
+
+We can create a dropdown selection widget via `alt.binding_select`.
+Then, instead of binding `alt.selection_single` to the legend we can
+pass the dropdown to the `bind` parameter.
+
+The dropdown requires an array to be passed to the options parameter;
+here we sort the genres alphabetically before passing them to the
+dropdown.
+
+Since we recreated the `select_genre` variable, we need to also redefine
+the opacity encoding and add this new variable to overwrite the old one.
+
+Now you can use the dropdown to select movies of different genres!
+
+Note that the location of the dropdown is fixed and there is no way
+change its location.
+
+---
+
+## Setting the default value in a widget
+
+``` python
+select_genre = alt.selection_single(
+    fields=['Major_Genre'], bind=dropdown, init={'Major_Genre': 'Comedy'})
+
+points.add_selection(select_genre).encode(
+    opacity=alt.condition(select_genre, alt.value(0.7), alt.value(0.1)))
+```
+
+<iframe src="/module7/charts/03/unnamed-chunk-4.html" width="100%" height="420px" style="border:none;">
+</iframe>
+
+Notes: If we don’t specify a default value in a widget, the first value
+in the options array show up in the widget, but it is not used for
+selection until we interact with the dropdown.
+
+If we wanted to set a specific default value and immediately use it to
+highlight the data, we could specify which column and value to use via
+the `init` parameter as we have done in this slide.
+
+---
+
+## Adding multiple widgets to the same charts
+
+``` python
+mpaa_rating = sorted(movies['MPAA_Rating'].unique())
+dropdown_mpaa = alt.binding_radio(name='MPAA Rating ', options=mpaa_rating)
+dropdown_genre = alt.binding_select(name='Genre ', options=genres)
+select_genre_and_mpaa = alt.selection_single(
+    fields=['Major_Genre', 'MPAA_Rating'],
+    bind={'Major_Genre': dropdown_genre, 'MPAA_Rating': dropdown_mpaa})
+
+points.add_selection(select_genre_and_mpaa).encode(
+    opacity=alt.condition(select_genre_and_mpaa, alt.value(0.7), alt.value(0.1)))
```

Comment:
I'll put this on wish list for now and revisit when I am down with all other changes. In general I like showing all the code if there is room, but also see value in being consistent between the slide and highlighting what is new.

Reply:
I tried a bit with this now, but I can't find a good way of reusing all the code. The code that would be reused is also something they have seen multiple times now: x, y, color, so they should be very used to that. One slide deck has been spent on condition already, and we do show when we change that. I also think the amount of code if we reuse will be confusing and it will be hard to see what is new, especially when flipping through slides on their own.